x-pack/plugins/cases/server/services/user_actions/abstract_builder.ts

```diff
@@ -19,17 +20,21 @@ import { ActionTypes, NONE_CONNECTOR_ID } from '../../../common/api';
 import type {
   BuilderDeps,
   BuilderParameters,
-  BuilderReturnValue,
   CommonBuilderArguments,
+  PersistableUserActionFields,
+  UserActionLogBody,
   UserActionParameters,
 } from './types';
 import type { PersistableStateAttachmentTypeRegistry } from '../../attachment_framework/persistable_state_registry';
+import { PersistableUserAction } from './persistable_user_action';
 
 export abstract class UserActionBuilder {
   protected readonly persistableStateAttachmentTypeRegistry: PersistableStateAttachmentTypeRegistry;
+  private readonly auditLogger: AuditLogger;
```

Comment:
Since the security plugin is required the audit logger will not be optional. I'll update the Authorization audit logger to be required in a separate PR.